src/treemodel/nodelist.js

```diff
@@ -50,14 +51,18 @@ class NodeListText extends Text {
 	}
 
 	/**
-	 * Custom toJSON method to solve child-parent circular dependencies.
+	 * Custom toJSON method to solve child-parent circular dependencies when serializing NodeListText to JSON string.
 	 *
 	 * @returns {Object} Clone of this object with the parent property replaced with its name.
 	 */
 	toJSON() {
 		const json = clone( this );
 
-		json.parent = json.parent ? this.parent.name : null;
+		// Due to circular references we need to remove parent reference.
+		json.parent = this.parent ? this.parent.name : null;
```

Comment:
If this is not needed anymore I could drop this across the ticket as in delta case.